src/util/interpolate.ts

```diff
@@ -37,6 +38,10 @@ function number(from: number, to: number, t: number): number {
     return from + t * (to - from);
 }
 
+function projection(from: Projection, to: Projection, interpolation: number): ProjectionTransition {
```

Comment:
If we compare this method to other interpolation methods we see that the input from and to are the same type as the return value. 
I think we should try and keep that concept in mind and see what can be done to define the projection object better to support this notion.

Reply:
The projection is unique in this regards, because an `orange` return color can hold the information of being between `yellow` and `red`, but I don't have a way to encapsulate an intermediate between `mercator` and `globe` in the same fashion, that's why i return the minimal information `[globe, mercator, 0.8]`, which will allow gl js to calculate what this intermediate projection look like.

Reply:
Maybe we can define "pure" protection as one that have both from and to the same with 1 for "weight".
and have the interpolate value calc a new Projection with new from to and weight.

Reply:
I was a bit concerned it would be verbose, but let's try to see if it's at least possible to implement with this approach. I've updated the code to follow this pattern.

Reply:
```ts
let errors = validateProjection({validateSpec, value: {type: ['step', ['zoom'], ['literal',{'from': 'globe', 'to': 'globe', transition: 1}], 10, ['literal',{'from': 'mercator', 'to': 'mercator', transition: 1}]]}, styleSpec: v8, style: {} as any});

errors = validateProjection({validateSpec, value: {type: ['literal', {'from': 'mercator', 'to': 'mercator', transition: 1}]}, styleSpec: v8, style: {} as any});

errors = validateProjection({validateSpec, value: {type: ['interpolate-projection', ['linear'],['zoom'], 0, {'from': 'globe', 'to': 'globe', transition: 1}, 10, {'from': 'mercator', 'to': 'mercator', transition: 1}]}, styleSpec: v8, style: {} as any});

```

it's verbose. i'd really like if we can also allow `globe` instead of `['literal',{'from': 'globe', 'to': 'globe', transition: 1}]`